File: Classes/EVEOnlineAPI/Character/EVEMailBodies.m
//
//  EVEMailBodies.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 1/7/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "EVEMailBodies.h"


@implementation EVEMailBodiesItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"messageID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"text":@{@"type":@(EVEXMLSchemePropertyTypeString), @"elementName":@"_"}};
	return scheme;
}

@end


@implementation EVEMailBodies

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"messages":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMailBodiesItem class]},
				   @"missingMessageIDs":@{@"type":@(EVEXMLSchemePropertyTypeObject), @"class":[NSArray class], @"transformer":^(id value) {
					   if ([value length] == 0)
						   return (id) nil;
					   else
						   return [[value componentsSeparatedByString:@""] valueForKey:@"intValue"];
				   }}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEMailBodies.swift
//
//  EVEMailBodies.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 29.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEMailBodiesItem: EVEObject {
	public var messageID: Int64 = 0
	public var text: String = ""
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"messageID":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
			"text":EVESchemeElementType.String(elementName:nil, transformer:nil),
		]
	}
}


public class EVEMailBodies: EVEResult {
	public var messages: [EVEMailBodiesItem] = []
	public var missingMessageIDs: [Int] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"messages":EVESchemeElementType.Rowset(elementName: nil, type: EVEMailBodiesItem.self, transformer: nil),
			"missingMessageIDs":EVESchemeElementType.IntList(elementName: nil, transformer: nil),
		]
	}
}

